src/Microsoft.Extensions.DependencyInjection/ServiceLookup/ClosedIEnumerableService.cs

```diff
@@ -3,19 +3,18 @@
 
 using System;
 using System.Collections.Generic;
-using System.Linq;
 
 namespace Microsoft.Extensions.DependencyInjection.ServiceLookup
 {
     internal class ClosedIEnumerableService : IService
     {
-        private readonly Type _itemType;
         private readonly ServiceEntry _serviceEntry;
 
         public ClosedIEnumerableService(Type itemType, ServiceEntry entry)
         {
-            _itemType = itemType;
+            ServiceType = itemType;
             _serviceEntry = entry;
+            ImplementationType = itemType.MakeArrayType();
```

Comment:
I can do that in another PR if you are worried about the overhead of calling a mostly null delegate in a method that'll be elided 😄 , there's enough churn here already and this adds nothing.